### PR TITLE
Remove memset for PROBLEME_HEBDO, de-couple MPS writer a bit

### PR DESCRIPTION
**Pull request by flomnes**
* Remove segfault-prone `memset` on structure `PROBLEME_HEBDO`
* Initialize members instead
* De-couple the MPS Writer factory class from god-class `PROBLEME_HEBDO` (opportunism on my part)

### CODE REVIEW

File: src/solver/simulation/sim_structure_probleme_economique.h [hunk 1]
```diff
@@ -501,100 +499,100 @@ struct PROBLEME_HEBDO
...
-    VALEURS_DE_NTC_ET_RESISTANCES** ValeursDeNTCRef;
+    VALEURS_DE_NTC_ET_RESISTANCES** ValeursDeNTC = nullptr;
+    VALEURS_DE_NTC_ET_RESISTANCES** ValeursDeNTCRef = nullptr;
 
-    int NombreDePasDeTemps;
-    int NombreDePasDeTempsRef;
-    int* NumeroDeJourDuPasDeTemps;
+    int NombreDePasDeTemps = 0;
+    int NombreDePasDeTempsRef = 0;
+    int* NumeroDeJourDuPasDeTemps = 0;
```
Comment (flomnes) [inline src/solver/simulation/sim_structure_probleme_economique.h]: Not sure if these initializations really make sense. TBD.

Comment (JasonMarechal25) [reply src/solver/simulation/sim_structure_probleme_economique.h]: Better safe than sorry

File: src/solver/optimisation/opt_calcul_des_pmin_MUT_MDT.cpp [hunk 2]
```diff
@@ -35,6 +35,8 @@
 #include <yuni/io/file.h>
 #include "opt_fonctions.h"
 
+#define ZERO_PMIN 1.e-2
```
Comment (JasonMarechal25) [inline src/solver/optimisation/opt_calcul_des_pmin_MUT_MDT.cpp]: Is it possile to use const expression (const or constxpr) instead of define ?

Comment (flomnes) [reply src/solver/optimisation/opt_calcul_des_pmin_MUT_MDT.cpp]: It should be, will look into it.

Comment (sylvlecl) [reply src/solver/optimisation/opt_calcul_des_pmin_MUT_MDT.cpp]: Suggestion to rename headers to `*_constants.h` accordingly

Comment (flomnes) [reply src/solver/optimisation/opt_calcul_des_pmin_MUT_MDT.cpp]: > Suggestion to rename headers to `*_constants.h` accordingly

[Done](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1146/commits/3f277ec9809fbb8741b869bd08cb817027ac0791)

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**